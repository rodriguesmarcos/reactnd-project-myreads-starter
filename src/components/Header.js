import React from 'react';
import { Link } from 'react-router-dom';

const Header = ({location}) => (
  <div className="header list-books-title">
    { location.pathname !== "/" && (
      <Link
        className="close-search"
        to="/">
        Close</Link>
    )}
    <h1>MyReads</h1>
  </div>
);

export default Header;
