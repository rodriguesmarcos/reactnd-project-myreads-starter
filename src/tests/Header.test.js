import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../components/Header.js';

describe('<Header />', () => {

  const minProps = {
    location: { pathname: "/" }
  }

  const notHome = {
    location: { pathname: "/search" }
  }

  test( 'renders without exploding', () => {
    const wrapper = shallow(<Header {...minProps}/>);
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test( 'renders back button', () => {
    const wrapper = shallow(<Header {...notHome}/>);
    expect(wrapper.find('Link').length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
