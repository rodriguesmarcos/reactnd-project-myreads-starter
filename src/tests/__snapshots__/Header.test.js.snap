exports[`<Header /> renders back button 1`] = `
<div
  className="header list-books-title">
  <Link
    className="close-search"
    replace={false}
    to="/">
    Close
  </Link>
  <h1>
    MyReads
  </h1>
</div>
`;

exports[`<Header /> renders without exploding 1`] = `
<div
  className="header list-books-title">
  <h1>
    MyReads
  </h1>
</div>
`;
